modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts

```diff
@@ -49,12 +56,26 @@ const applyCellData = (editor: Editor, cells: HTMLTableCellElement[], data: Cell
   const dom = editor.dom;
   const isSingleCell = cells.length === 1;
 
-  Arr.each(cells, (cell) => {
+  const table = TableLookup.table(SugarElement.fromDom(cells[0])).getOrNull();
+
+  const warehouse = Warehouse.Warehouse.fromTable(table);
+
+  const allCells = Warehouse.Warehouse.justCells(warehouse);
+  const columns = Warehouse.Warehouse.justColumns(warehouse);
+
+  const selectedCells = Arr.filter(allCells, (cellA) =>
+    Arr.exists(cells, (cellB) =>
+      cellA.element.dom === cellB
```

Comment:
This will never match, as one is a sugar element and the other is a dom element. Also you should be using `Compare.eq` here instead of comparing by dom elements.

Reply:
No, `cellA.element` is a sugar-element and cellB is an ordinary HTMLTableCellElement.
I'll change it to Compare.eq though.

Reply:
Right, that makes more sense. However they are being converted from sugar elements to dom elements in a previous step, so that could potentially be optimized.

Reply:
Seems sensible. No idea why that was done though.